$OpenBSD: patch-third-party_pjproject_source_pjlib_src_pj_ssl_sock_ossl_c,v 1.5 2018/05/03 21:10:49 sthen Exp $

we don't have SSL_get_shared_curve, SSL_set1_curves, SSL_CTX_set1_curves,
SSL_set1_{client_,}sigalgs_list etc.

Index: third-party/pjproject/source/pjlib/src/pj/ssl_sock_ossl.c
--- third-party/pjproject/source/pjlib/src/pj/ssl_sock_ossl.c.orig
+++ third-party/pjproject/source/pjlib/src/pj/ssl_sock_ossl.c
@@ -53,7 +53,7 @@
 #include <openssl/rand.h>
 #include <openssl/opensslconf.h>
 
-#if !defined(OPENSSL_NO_EC) && OPENSSL_VERSION_NUMBER >= 0x1000200fL
+#if !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_NO_EC) && OPENSSL_VERSION_NUMBER >= 0x1000200fL
 
 #   include <openssl/obj_mac.h>
 
@@ -600,7 +600,7 @@ static pj_status_t init_openssl(void)
 
 	SSL_set_session(ssl, SSL_SESSION_new());
 
-#if !defined(OPENSSL_NO_EC) && OPENSSL_VERSION_NUMBER >= 0x1000200fL
+#if !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_NO_EC) && OPENSSL_VERSION_NUMBER >= 0x1000200fL
 	openssl_curves_num = SSL_get_shared_curve(ssl,-1);
 	if (openssl_curves_num > PJ_ARRAY_SIZE(openssl_curves))
 	    openssl_curves_num = PJ_ARRAY_SIZE(openssl_curves);
@@ -1229,7 +1229,7 @@ static pj_status_t set_cipher_list(pj_ssl_sock_t *ssoc
 
 static pj_status_t set_curves_list(pj_ssl_sock_t *ssock)
 {
-#if !defined(OPENSSL_NO_EC) && OPENSSL_VERSION_NUMBER >= 0x1000200fL
+#if !defined(LIBRESSL_VERSION_NUMBER) && !defined(OPENSSL_NO_EC) && OPENSSL_VERSION_NUMBER >= 0x1000200fL
     int ret;
     int curves[PJ_SSL_SOCK_MAX_CURVES];
     unsigned cnt;
@@ -1260,7 +1260,7 @@ static pj_status_t set_curves_list(pj_ssl_sock_t *ssoc
 
 static pj_status_t set_sigalgs(pj_ssl_sock_t *ssock)
 {
-#if OPENSSL_VERSION_NUMBER >= 0x1000200fL
+#if !defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x1000200fL
     int ret;
 
     if (ssock->param.sigalgs.ptr && ssock->param.sigalgs.slen) {
